![imdb logo](imdb%20logo.jpg)

# **Introduction**
I have access to the IMDb Movies dataset, which is structured in a long format based on movie genres. This dataset contains a multitude of insightful details, including movie titles, release dates, runtimes, ratings, voter counts, approval indices, budgets, domestic and international revenues, revenue disparities, and a critical indicator that distinguishes between movie profitability and losses.

In my analysis, conducted within the SQL environment of Jupyter Notebook, I will embark on a journey to explore the intricate relationship between movie genres and various aspects of the film industry. By utilizing SQL queries and advanced data manipulation techniques, I aim to uncover valuable patterns, correlations, and statistics that can provide deep insights into the world of cinema. My investigation will delve into genre-specific trends, profitability across different genres, audience preferences, budget allocation, and much more.

Join me as I navigate the cinematic landscape, guided by data, and reveal the captivating narratives concealed within the IMDb Movies dataset, now presented in its long data format.

The data visualization dashboard (Tableau) and dataset can be accessed at the following link:
1. Tableau Public : https://tinyurl.com/moviesimdbdashbord
2. Dataset : https://tinyurl.com/imdbdatasets

- On this occasion, I will try to display 10 film data :

In [1]:
SELECT * FROM 'imdb_movies_dataset - long_data_genre.csv'
LIMIT 10;

movie_title    genre  ... difference_of_revenue  indicator
0              80 for Brady (2023)   Comedy  ...              47376000     Profit
1              80 for Brady (2023)    Drama  ...              47376000     Profit
2              80 for Brady (2023)    Sport  ...              47376000     Profit
3         A Man Called Otto (2022)   Comedy  ...             116804606     Profit
4         A Man Called Otto (2022)    Drama  ...             116804606     Profit
5                 Amsterdam (2022)   Comedy  ...             -35651205       Loss
6                 Amsterdam (2022)    Drama  ...             -35651205       Loss
7                 Amsterdam (2022)  History  ...             -35651205       Loss
8           Armageddon Time (2022)    Drama  ...              -7063064       Loss
9  Avatar: The Way of Water (2022)   Action  ...            2473765808     Profit

[10 rows x 13 columns]

Based on the data above, it can be seen that there are many duplicate titles, but they are segmented based on different genres.

In the first analysis, I will perform a query to display the films with the highest ratings in each year :

In [2]:
SELECT
	i.year AS year,
	MAX(i.rating) AS max_rating
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
GROUP BY i.year
ORDER BY year DESC;

year  max_rating
0   2023         5.2
1   2022         8.3
2   2021         8.2
3   2020         8.2
4   2019         8.4
..   ...         ...
91  1929         5.6
92  1925         7.9
93  1920         4.3
94  1916         6.1
95  1915         6.2

[96 rows x 2 columns]

Next, I will focus on analyzing films based on title, genre and rating.
First, I will show the top 10 films by rating without including genre segmentation :

In [3]:
SELECT
	i.movie_title AS title,
	ROUND(AVG(i.rating),1) AS rating,
	AVG(i.number_of_voters) AS voters_total,
	AVG(i.approval_index) AS approval_total
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
GROUP BY i.movie_title
ORDER BY rating DESC
LIMIT 10;

title  ...  approval_total
0                    The Shawshank Redemption (1994)  ...       10.000000
1                               The Godfather (1972)  ...        9.643790
2                             The Dark Knight (2008)  ...        9.666757
3                                    Solitude (2005)  ...        1.661377
4  The Lord of the Rings: The Return of the King ...  ...        9.426391
5                                Pulp Fiction (1994)  ...        9.391076
6                                   Inception (2010)  ...        9.371470
7       The Lord of the Rings: The Two Towers (2002)  ...        9.147916
8  The Lord of the Rings: The Fellowship of the R...  ...        9.224165
9                                  Fight Club (1999)  ...        9.306151

[10 rows x 4 columns]

Based on the query results above, the 3 films with the highest ratings are:
1. The Shawshank Redemption (1994)
2. The Godfather (1972)
3. The Lord of the Rings: The Return of the King (2003)

### **The Shawshank Redemption (1994)**
The Shawshank Redemption (1994) has the highest rating, namely 9.3. This film is a classic film directed by Frank Darabont. It falls under the genre of drama. This film is renowned for its compelling storyline, memorable characters, and its exploration of themes such as hope and redemption within the confines of a prison setting.


### **The Godfather (1972)**
The Godfather (1972) came in second with a 9.2 rating. This film is a classic crime drama film directed by Francis Ford Coppola and based on the novel of the same name by Mario Puzo. The film revolves around the Corleone family, a powerful Italian-American mafia clan, and their patriarch, Vito Corleone, played by Marlon Brando. The film explores the themes of power, family, loyalty, and the consequences of a life in organized crime. It also spawned two sequels, "The Godfather Part II" (1974) and "The Godfather Part III" (1990), which further explore the Corleone family's saga.


### **The Lord of the Rings: The Return of the King (2003)**
This film is in third place with a rating of 9, following other films below it. The Lord of the Rings: The Return of the King" is a fantasy film released in 2003, directed by Peter Jackson. It is the third and final installment of "The Lord of the Rings" film trilogy, based on the epic fantasy novel" The Return of the King" by J.R.R. Tolkien. The film concludes the epic tale set in the fictional realm of Middle-earth.

After I succeeded in getting the 10 films with the highest ratings, and a brief explanation of the 3 films above, on this occasion, I will try to analyze which film genres are most popular with film audiences. 

I will group some of these genres because there are formatting errors in the dataset.

In [4]:
SELECT
	CASE WHEN i.genre = 'Film-Noir' OR i.genre = 'News' OR i.genre = 'Criminal' THEN 'Others'
		 ELSE i.genre END AS movies_genre,
	ROUND(AVG(i.number_of_voters),0) AS voters_number,
	COUNT(*) AS total_number_of_productions
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
GROUP BY i.genre
ORDER BY voters_number DESC, total_number_of_productions DESC;

movies_genre  voters_number  total_number_of_productions
0        Sci-Fi       271237.0                          378
1     Adventure       200065.0                          966
2        Action       196265.0                         1206
3     Animation       164876.0                          240
4       Fantasy       160763.0                          346
5       Mystery       152757.0                          408
6      Thriller       147413.0                          661
7         Crime       144504.0                          776
8           War       134574.0                           85
9         Drama       129363.0                         2176
10    Biography       123083.0                          338
11       Comedy       104712.0                         1641
12       Horror       104117.0                          448
13      Western       101940.0                           44
14       Family        94033.0                          261
15      History        93605.0                          149
16      Romance        88317.0                          717
17       Others        83892.0                            1
18        Sport        82530.0                          114
19       Others        73159.0                            7
20        Music        69580.0                          145
21      Musical        66656.0                           54
22  Documentary        24256.0                           58
23       Others           53.0                            1

Based on the query results above, it can be seen that the top 3 films that are most popular with audiences are films with the genre:
1. Sci-Fi (271,237 Votes)
2. Action (200,065 Votes)
3. Adventure (196,265 Votes)

Based on this, I will try to find the 20 best films from 3 genre based on the number of votes on IMDB.

# **Sci-Fi Movies**

In [5]:
SELECT
	i.movie_title,
	i.genre,
	i.number_of_voters AS voters_number
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
WHERE i.genre = 'Sci-Fi'
ORDER BY voters_number DESC
LIMIT 10;

movie_title   genre  voters_number
0                              Inception (2010)  Sci-Fi        2368570
1                             The Matrix (1999)  Sci-Fi        1924457
2                           Interstellar (2014)  Sci-Fi        1852238
3                           The Avengers (2012)  Sci-Fi        1398299
4                           The Prestige (2006)  Sci-Fi        1342184
5                     Back to the Future (1985)  Sci-Fi        1214789
6             Terminator 2: Judgment Day (1991)  Sci-Fi        1106289
7                 Avengers: Infinity War (2018)  Sci-Fi        1091968
8                               Iron Man (2008)  Sci-Fi        1066239
9  Eternal Sunshine of the Spotless Mind (2004)  Sci-Fi        1015223

# **Action Movies**

In [6]:
SELECT
	i.movie_title,
	i.genre,
	i.number_of_voters AS voters_number
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
WHERE i.genre = 'Action'
ORDER BY voters_number DESC
LIMIT 10;

movie_title   genre  voters_number
0                             The Dark Knight (2008)  Action        2669915
1                                   Inception (2010)  Action        2368570
2                                  The Matrix (1999)  Action        1924457
3  The Lord of the Rings: The Fellowship of the R...  Action        1886597
4  The Lord of the Rings: The Return of the King ...  Action        1857170
5                       The Dark Knight Rises (2012)  Action        1715219
6       The Lord of the Rings: The Two Towers (2002)  Action        1677003
7                                   Gladiator (2000)  Action        1510184
8                               Batman Begins (2005)  Action        1480348
9                                The Avengers (2012)  Action        1398299

# **Adventure Movies**

In [7]:
SELECT
	i.movie_title,
	i.genre,
	i.number_of_voters AS voters_number
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
WHERE i.genre = 'Adventure'
ORDER BY voters_number DESC
LIMIT 10;

movie_title      genre  voters_number
0                                   Inception (2010)  Adventure        2368570
1  The Lord of the Rings: The Fellowship of the R...  Adventure        1886597
2  The Lord of the Rings: The Return of the King ...  Adventure        1857170
3                                Interstellar (2014)  Adventure        1852238
4       The Lord of the Rings: The Two Towers (2002)  Adventure        1677003
5                                   Gladiator (2000)  Adventure        1510184
6                        Inglourious Basterds (2009)  Adventure        1461682
7                                      Avatar (2009)  Adventure        1316701
8                          Back to the Future (1985)  Adventure        1214789
9                     Guardians of the Galaxy (2014)  Adventure        1187949

After conducting analysis related to films, next I will focus on trends in the number of film productions, analysis of budget and revenue for films in the following dataset.

The first analysis, I will aggregate the data to display the number of film productions, average number of voters and average of approval index based on genre from 2000 to the present, in hierarchical form :

In [8]:
SELECT
	i.year AS year,
	CASE WHEN i.genre IS NULL THEN 'Total of Calculation ='
		ELSE i.genre END AS genre,
	COUNT(i.genre) AS number_of_genre,
	ROUND(AVG(i.number_of_voters),2) AS number_of_voters,
	ROUND(AVG(i.approval_index),2) AS number_approval
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
GROUP BY ROLLUP (i.year,i.genre)
HAVING (i.year) > '2000'
ORDER BY year DESC, genre ASC;

year                   genre  ...  number_of_voters  number_approval
0    2023                  Comedy  ...            359.00             2.12
1    2023                   Drama  ...            359.00             2.12
2    2023                   Sport  ...            359.00             2.12
3    2023  Total of Calculation =  ...            359.00             2.12
4    2022                  Action  ...         185202.71             5.29
..    ...                     ...  ...               ...              ...
454  2001                   Sport  ...          48285.50             4.53
455  2001                Thriller  ...         149643.76             5.19
456  2001  Total of Calculation =  ...         124889.71             4.82
457  2001                     War  ...         156699.00             6.49
458  2001                 Western  ...          16797.00             4.09

[459 rows x 5 columns]

Based on the query results above, there are very few film genres produced in 2023, namely only 3. Because this film dataset is an update of films as of February 2023.

The following SQL code produces a report that aggregates data by year and genre. This report will include some statistical information about the films in the dataset that have a release year greater than 2000.

The output results also display the average number of voters and the average approval index for each combination of year and genre.

I want to show the films with the highest approval index in each genre :

In [9]:
SELECT 
	genre, 
	movie_title,
	approval_index
FROM (
    	SELECT 
			genre, 
			movie_title, 
			approval_index, 
			ROW_NUMBER() OVER (PARTITION BY genre ORDER BY approval_index DESC) AS rn
    	FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
) AS ranked
WHERE rn = 1;


genre  ... approval_index
0     Animation  ...       8.548207
1          News  ...       6.040821
2     Biography  ...       8.811903
3        Horror  ...       8.434864
4   Documentary  ...       6.865744
5       Musical  ...       7.447506
6       Mystery  ...       8.932272
7         Music  ...       8.424185
8       History  ...       8.432932
9        Comedy  ...       8.629912
10       Family  ...       7.641060
11        Crime  ...       9.666757
12      Romance  ...       9.291674
13      Western  ...       8.684168
14     Thriller  ...       8.841443
15        Drama  ...      10.000000
16    Adventure  ...       9.426391
17    Film-Noir  ...       6.925697
18        Sport  ...       7.882077
19          War  ...       8.822924
20       Action  ...       9.666757
21      Fantasy  ...       8.781104
22     Criminal  ...       1.416379
23       Sci-Fi  ...       9.371470

[24 rows x 3 columns]

I want to show the films with the highest number of votes in each genre:

In [10]:
SELECT 
	genre, 
	movie_title,
	number_of_voters
FROM (
    	SELECT 
			i.genre, 
			i.movie_title, 
			i.number_of_voters, 
			ROW_NUMBER() OVER (PARTITION BY i.genre ORDER BY i.number_of_voters DESC) AS rn
    	FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
) AS ranked
WHERE rn = 1;


genre  ... number_of_voters
0     Animation  ...          1066010
1          News  ...            83892
2     Biography  ...          1426046
3        Horror  ...          1030407
4   Documentary  ...           145581
5       Musical  ...           286676
6       Mystery  ...          1664484
7         Music  ...           874250
8       History  ...          1043556
9        Comedy  ...          1426046
10       Family  ...           643366
11        Crime  ...          2669915
12      Romance  ...          2093927
13      Western  ...          1566971
14     Thriller  ...          1441909
15        Drama  ...          2695887
16    Adventure  ...          2368570
17    Film-Noir  ...           139012
18        Sport  ...           691828
19          War  ...          1461682
20       Action  ...          2669915
21      Fantasy  ...          1316701
22     Criminal  ...               53
23       Sci-Fi  ...          2368570

[24 rows x 3 columns]

If you look closely, films that have a high approval index tend to have a high number of voters too. On this occasion I will try to find the correlation between these 2 columns for each genre :

In [11]:
WITH GenreStats AS (
    SELECT
        i.genre,
        AVG(i.approval_index) AS avg_approval_index,
        AVG(i.number_of_voters) AS avg_number_of_voters
    FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
    GROUP BY i.genre
)
SELECT
    i.genre,
    ROUND(CORR(approval_index, number_of_voters),3) AS correlation
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
INNER JOIN GenreStats
ON i.genre = GenreStats.genre
GROUP BY i.genre;

genre  correlation
0        Comedy        0.732
1         Drama        0.702
2         Sport        0.676
3       History        0.719
4        Action        0.755
5     Adventure        0.758
6       Fantasy        0.746
7        Horror        0.774
8       Mystery        0.772
9      Thriller        0.761
10        Crime        0.725
11    Biography        0.745
12        Music        0.704
13       Family        0.729
14       Sci-Fi        0.794
15    Animation        0.782
16      Romance        0.668
17      Musical        0.732
18          War        0.749
19      Western        0.664
20  Documentary        0.614
21         News          NaN
22     Criminal          NaN
23    Film-Noir        0.939

Based on the results of the correlation analysis above, it can be seen that each film genre has a positive correlation between the approval index and the number of voters, so it can be concluded that the higher the approval index, the higher the number of voters.

The next analysis is that I will group film categories based on the length of the film into the following 3 categories:

1. Epic Film : runtime more than 180 minutes
2. Feature Film : runtime between 70 - 179 minutes
3. Short Movie : runtime between 30 - 69 minutes

In [12]:
SELECT
	i.movie_title AS title,
	ROUND(AVG(i.runtime),0) AS duration,
	CASE WHEN duration > 180 THEN 'Epic Movie'
		 WHEN duration > 70 THEN 'Feature Movie'
		 WHEN duration > 30 THEN 'Short Movie' END AS duration_category
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
GROUP BY i.movie_title
ORDER BY duration DESC;

title  duration duration_category
0                       Gettysburg (1993)     271.0        Epic Movie
1     The Greatest Story Ever Told (1965)     260.0        Epic Movie
2                           Hamlet (1996)     242.0        Epic Movie
3               Gone with the Wind (1939)     238.0        Epic Movie
4      Once Upon a Time in America (1984)     229.0        Epic Movie
...                                   ...       ...               ...
4307          The Land Before Time (1988)      69.0       Short Movie
4308                         Bambi (1942)      69.0       Short Movie
4309        Pooh's Heffalump Movie (2005)      68.0       Short Movie
4310            She Done Him Wrong (1933)      66.0       Short Movie
4311               Winnie the Pooh (2011)      63.0       Short Movie

[4312 rows x 3 columns]

In [13]:
WITH summary AS (
		SELECT
			i.movie_title AS title,
			ROUND(AVG(i.runtime),0) AS duration,
			CASE WHEN duration > 180 THEN 'Epic Movie'
				 WHEN duration > 70 THEN 'Feature Movie'
				 WHEN duration > 30 THEN 'Short Movie' END AS duration_category
		FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
		GROUP BY i.movie_title
		ORDER BY duration DESC)
SELECT
	duration_category,
	COUNT(*) AS number_of_time_category
FROM summary
GROUP BY duration_category;

duration_category  number_of_time_category
0     Feature Movie                     4262
1       Short Movie                        5
2        Epic Movie                       45

Based on the results above, I used a common table expression to query the duration category. It turns out that the film industry produces more feature films with a duration of around 70 to 180 minutes.

Film producers realize that a film that has a detailed plot is not necessarily a film that will attract the interest of the audience. In fact, films with a tight plot and the right length of film are really paid attention to by film producers, so it's not surprising that many film scenes are cut.

The next analysis is that I want to carry out an analysis of the budget spent by the producer on the film.

The first query will display the sub average budget, domestic revenue and international revenue for each genre per year (2020 - 2023) using the ROLL UP operator

In [14]:
SELECT
	i.year AS year,
	CASE WHEN i.genre IS NULL THEN 'Total of Calculation ='
		ELSE i.genre END AS genre,
	COUNT(i.genre) AS number_of_genre,
	ROUND(AVG(i.budget),2) AS average_budget,
	ROUND(AVG(i.domestic_revenue),2) AS average_dom_revenue,
	ROUND(AVG(i.international_revenue),2) AS average_intl_revenue
FROM 'imdb_movies_dataset - long_data_genre.csv' AS i
GROUP BY ROLLUP (i.year,i.genre)
HAVING (i.year) >= 2020
ORDER BY year DESC, genre ASC;

year                   genre  ...  average_dom_revenue  average_intl_revenue
0   2023                  Comedy  ...         3.768800e+07          3.768800e+07
1   2023                   Drama  ...         3.768800e+07          3.768800e+07
2   2023                   Sport  ...         3.768800e+07          3.768800e+07
3   2023  Total of Calculation =  ...         3.768800e+07          3.768800e+07
4   2022                  Action  ...         1.911612e+08          4.580530e+08
5   2022               Adventure  ...         2.044474e+08          5.356095e+08
6   2022               Animation  ...         1.146508e+08          2.705666e+08
7   2022               Biography  ...         8.606836e+07          1.513545e+08
8   2022                  Comedy  ...         9.751296e+07          2.182305e+08
9   2022                   Crime  ...         1.450919e+08          3.089821e+08
10  2022                   Drama  ...         1.171549e+08          2.321054e+08
11  2022                  Family  ...         9.585084e+07          4.045601e+08
12  2022                 Fantasy  ...         3.357912e+08          1.003496e+09
13  2022                 History  ...         3.254252e+07          6.187534e+07
14  2022                  Horror  ...         6.117251e+07          1.044718e+08
15  2022                   Music  ...         1.510400e+08          2.809027e+08
16  2022                 Mystery  ...         7.133240e+07          1.219169e+08
17  2022                 Romance  ...         7.925337e+07          1.543059e+08
18  2022                  Sci-Fi  ...         1.730629e+08          4.113336e+08
19  2022                Thriller  ...         5.126587e+07          9.191817e+07
20  2022  Total of Calculation =  ...         1.324405e+08          3.110549e+08
21  2021                  Action  ...         1.128765e+08          3.096447e+08
22  2021               Adventure  ...         1.263350e+08          3.087896e+08
23  2021               Animation  ...         7.922808e+07          1.801034e+08
24  2021               Biography  ...         2.480864e+07          6.417432e+07
25  2021                  Comedy  ...         6.328078e+07          1.391960e+08
26  2021                   Crime  ...         3.974569e+07          1.239529e+08
27  2021                   Drama  ...         2.179319e+07          5.173874e+07
28  2021                  Family  ...         4.822181e+07          1.157849e+08
29  2021                 Fantasy  ...         2.008109e+08          4.397460e+08
30  2021                 History  ...         8.165977e+06          1.856698e+07
31  2021                  Horror  ...         3.253876e+07          6.576034e+07
32  2021                 Musical  ...         3.425274e+07          5.923665e+07
33  2021                 Mystery  ...         2.460029e+07          6.041874e+07
34  2021                 Romance  ...         1.706446e+07          3.066662e+07
35  2021                  Sci-Fi  ...         5.818028e+07          1.780805e+08
36  2021                   Sport  ...         7.806911e+06          1.923518e+07
37  2021                Thriller  ...         7.930010e+07          3.031820e+08
38  2021  Total of Calculation =  ...         7.030889e+07          1.805189e+08
39  2020                  Action  ...         1.718399e+07          6.028085e+07
40  2020               Adventure  ...         3.434664e+07          9.460886e+07
41  2020               Animation  ...         4.164492e+07          1.176529e+08
42  2020               Biography  ...         1.041000e+07          1.670375e+07
43  2020                  Comedy  ...         2.540835e+07          6.737666e+07
44  2020                   Crime  ...         9.500530e+05          1.087439e+06
45  2020                   Drama  ...         1.422820e+07          2.641354e+07
46  2020                  Family  ...         5.355578e+07          1.352264e+08
47  2020                 Fantasy  ...         2.593824e+07          7.119638e+07
48  2020                  Horror  .

In the following analysis, I will try to show the films with the highest budgets for each genre :

In [15]:
SELECT genre, movie_title, budget
FROM (SELECT
		i.genre,
		i.movie_title,
		i.budget,
		ROW_NUMBER () OVER (PARTITION BY i.genre ORDER BY i.budget DESC) AS rn
		FROM 'imdb_movies_dataset - long_data_genre.csv' AS i) AS ranked
WHERE rn = 1;

genre                                    movie_title     budget
0     Animation                                 Tangled (2010)  260000000
1          News                   An Inconvenient Truth (2006)    1000000
2     Biography                               Alexander (2004)  155000000
3        Horror                             World War Z (2013)  190000000
4   Documentary                              Jackass 3D (2010)   20000000
5       Musical                         West Side Story (2021)  100000000
6       Mystery                       Quantum of Solace (2008)  230000000
7         Music                                   Elvis (2022)   85000000
8       History                       Deepwater Horizon (2016)  156000000
9        Comedy                                 Tangled (2010)  260000000
10       Family  Harry Potter and the Half-Blood Prince (2009)  250000000
11        Crime                 The Fate of the Furious (2017)  250000000
12      Romance                                 Titanic (1997)  200000000
13      Western                         The Lone Ranger (2013)  225000000
14     Thriller                                 Spectre (2015)  300000000
15        Drama                       Avengers: Endgame (2019)  400000000
16    Adventure                Avatar: The Way of Water (2022)  460000000
17    Film-Noir                               Notorious (1946)    2000000
18        Sport                                     Ali (2001)  109000000
19          War                             Windtalkers (2002)  115000000
20       Action                Avatar: The Way of Water (2022)  460000000
21      Fantasy                Avatar: The Way of Water (2022)  460000000
22     Criminal                          The Good Thief (2003)   30000000
23       Sci-Fi                 Avengers: Age of Ultron (2015)  365000000

In the following analysis, I will try to show the films with the highest international revenue for each year :

In [16]:
SELECT 
	year, 
	movie_title, 
	international_revenue
FROM (SELECT
		i.year,
		i.movie_title,
		i.international_revenue,
		ROW_NUMBER () OVER (PARTITION BY i.year ORDER BY i.international_revenue DESC) AS rn
		FROM 'imdb_movies_dataset - long_data_genre.csv' AS i) AS ranked
WHERE rn = 1
ORDER BY year DESC;

year                            movie_title  international_revenue
0   2023                    80 for Brady (2023)               37688000
1   2022        Avatar: The Way of Water (2022)             2265935552
2   2021         Spider-Man: No Way Home (2021)             1910048245
3   2020                           Tenet (2020)              360240189
4   2019               Avengers: Endgame (2019)             2794731755
..   ...                                    ...                    ...
91  1929             The Broadway Melody (1929)                4358000
92  1925                  The Big Parade (1925)               22000000
93  1920  Over the Hill to the Poorhouse (1920)                3000000
94  1916    20,000 Leagues Under the Sea (1916)                8000000
95  1915           The Birth of a Nation (1915)               11000000

[96 rows x 3 columns]

Percobaan Line Chart


That's all for the project portfolio regarding IMDB film data analysis, hopefully it can provide useful insights, information and recommendations for watching films



Feby Renaldi (Data Analytics Enthusiast)